File: solutions/java/bird-watcher/1/src/main/java/BirdWatcher.java

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return new int[] { 0, 2, 5, 3, 7, 8, 4 };
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1]++;
    }

    public boolean hasDayWithoutBirds() {
        for (int numBirds : birdsPerDay) {
	    if (numBirds == 0)
		return true;
	}
	return false;
    }

    public int getCountForFirstDays(int numberOfDays) {
	int sum = 0;
        for (int i = 0; i < Math.min(numberOfDays, birdsPerDay.length); i++) {
	    sum += birdsPerDay[i];
	}
	return sum;
    }

    public int getBusyDays() {
        int numBusyDays = 0;
	for (int numBirds : birdsPerDay) {
	    if (numBirds >= 5)
		numBusyDays++;
	}
	return numBusyDays;
    }
}

File: solutions/csharp/bird-watcher/1/BirdWatcher.cs
using System;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => new int[] {0, 2, 5, 3, 7, 8, 4};

    public int Today() => birdsPerDay[birdsPerDay.Length - 1];

    public void IncrementTodaysCount() => birdsPerDay[birdsPerDay.Length - 1]++;

    public bool HasDayWithoutBirds() => Array.FindIndex(birdsPerDay, n => n==0) != -1;

    public int CountForFirstDays(int numberOfDays)
    {
        int sum = 0;
        for (int i = 0; i < Math.Min(numberOfDays, birdsPerDay.Length); i++)
        {
            sum += birdsPerDay[i];
        }
        return sum;
    }

    public int BusyDays() => Array.FindAll(birdsPerDay, n => n >= 5).Length;
}
